uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-action-dropdown.js

```diff
@@ -0,0 +1,172 @@
+/* eslint-env amd */
+
+define([
+  'common/lodash',
+  'common/moment',
+  'leave-absences/shared/modules/components',
+  'common/services/hr-settings'
+], function (_, moment, components) {
+  components.component('leaveRequestActionDropdown', {
+    bindings: {
+      request: '<',
+      requestStatuses: '<',
+      absenceTypes: '<',
+      /**
+       * Role is not a permission level in this case.
+       * For example Manager can act as Staff
+       * and Admin can act as either Manager or Staff.
+       */
+      role: '<'
+    },
+    templateUrl: ['shared-settings', function (sharedSettings) {
+      return sharedSettings.sharedPathTpl + 'components/leave-request-action-dropdown.html';
+    }],
+    controllerAs: 'vm',
+    controller: ['$log', '$q', '$rootScope', 'OptionGroup', 'dialog', 'shared-settings', controller]
+  });
+
+  function controller ($log, $q, $rootScope, OptionGroup, dialog, sharedSettings) {
+    $log.debug('Component: leave-request-action-dropdown');
+
+    var vm = this;
+
+    vm.actions = [];
+    vm.actionLabels = {
+      edit: 'Edit',
+      cancel: 'Cancel',
+      delete: 'Delete',
+      view: 'View',
+      respond: 'Respond',
+      approve: 'Approve',
+      reject: 'Reject'
+    };
+
+    /**
+     * Performs an action on a given leave request
+     *
+     * @param {LeaveRequestInstance} leaveRequest
+     * @param {string} action
+     */
+    vm.act = function (action) {
+      var map = {
+        approve: {
+          title: 'Approval',
+          btnClass: 'success',
+          btnLabel: 'Approve',
+          msg: 'Please confirm approval'
+        },
+        cancel: {
+          title: 'Cancellation',
+          btnClass: 'danger',
+          btnLabel: 'Confirm',
+          msg: 'Please confirm cancellation'
+        },
+        delete: {
+          title: 'Deletion',
+          btnClass: 'danger',
+          btnLabel: 'Confirm',
+          msg: 'This cannot be undone'
+        },
+        reject: {
+          title: 'Rejection',
+          btnClass: 'warning',
+          btnLabel: 'Reject',
+          msg: 'Please confirm rejection'
+        }
+      };
+
+      dialog.open({
+        title: 'Confirm ' + map[action].title + '?',
+        copyCancel: 'Cancel',
+        copyConfirm: map[action].btnLabel,
+        classConfirm: 'btn-' + map[action].btnClass,
+        msg: map[action].msg,
+        onConfirm: function () {
+          return vm.request[action]()
+            .then(function () {
+              $rootScope.$emit('LeaveRequest::' + (action === 'delete' ? 'deleted' : 'edit'), vm.request);
+            });
+        }
+      });
+    };
+
+    (function init () {
+      vm.actions = getActions(vm.request, vm.role);
```

Comment:
i wonder if
```js
getActions()
```
is better. In the end both input and output are properties of the controller